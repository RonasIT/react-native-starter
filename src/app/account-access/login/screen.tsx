import { yupResolver } from '@hookform/resolvers/yup';
import Constants from 'expo-constants';
import { isEmpty } from 'lodash';
import React from 'react';
import { useForm } from 'react-hook-form';
import { ScrollView, View } from 'react-native';
import { Image, Keyboard } from 'react-native-ui-lib';
import { assets } from '@assets/index';
import { AppVersion } from '@shared/app-version';
import { AppButton } from '@shared/button';
import { useTranslation } from '@shared/i18n';
import { InputFormGroup } from '@shared/input-form-group';
import { AppScreen } from '@shared/screen';
import { AppText } from '@shared/text';
import { commonStyle, createStyles } from '@styles';
import { loginScreenFacade } from './facade';
import { LoginForm } from './shared/forms';

export function LoginScreen(): JSX.Element {
  const translate = useTranslation('ACCOUNT_ACCESS.LOGIN');
  const { isSubmitting } = loginScreenFacade;
  const appName = Constants.expoConfig.name;

  function formSubmitted(values: LoginForm): void {
    loginScreenFacade.authorize(values);
  }

  const form = useForm<LoginForm>({
    defaultValues: new LoginForm(),
    resolver: yupResolver(LoginForm.validationSchema)
  });
  const { handleSubmit, formState, control } = form;

  return (
    <AppScreen style={commonStyle.container}>
      <ScrollView contentContainerStyle={style.content}>
        <Image source={assets.brand.logo} style={style.logo} />
        <AppText style={style.title} variant='largest'>
          {translate('TEXT_TITLE', { value: appName })}
        </AppText>
        <InputFormGroup<LoginForm>
          label={translate('TEXT_LOGIN')}
          name='email'
          testID='email-input'
          autoCapitalize='none'
          keyboardType='email-address'
          control={control}
        />
        <InputFormGroup<LoginForm>
          isPassword={true}
          testID='password-input'
          label={translate('TEXT_PASSWORD')}
          name='password'
          control={control}
        />
        <View style={style.footer}>
          <AppButton
            isLoading={isSubmitting}
            disabled={!isEmpty(formState.errors) && formState.isSubmitted}
            testID='submit-button'
            label={translate('BUTTON_SUBMIT')}
            onPress={handleSubmit(formSubmitted)}
          />
        </View>
        <AppVersion />
        <Keyboard.KeyboardAwareInsetsView />
      </ScrollView>
    </AppScreen>
  );
}

const style = createStyles({
  content: {
    paddingTop: '2rem'
  },
  logo: {
    width: '3.5rem',
    height: '3.5rem'
  },
  title: {
    marginVertical: '2rem'
  },
  footer: {
    marginTop: '2rem'
  }
});
