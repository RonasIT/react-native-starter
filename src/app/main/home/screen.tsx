import React, { ReactElement } from 'react';
import { AppScreen } from '../../../libs/shared/ui/ui-kit/screen';
import { UsersList } from '../../../libs/users/features/list';

export function HomeScreen(): ReactElement {
  return (
    <AppScreen testID='home-screen'>
      <UsersList />
    </AppScreen>
  );
}
