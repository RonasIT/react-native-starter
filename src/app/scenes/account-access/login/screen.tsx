import React from 'react';
import { LoginForm } from '@libs/auth/features/login-form';
import { commonStyle } from '@libs/shared/ui/styles';
import { AppScreen } from '@libs/shared/ui/ui-kit/screen';

export function LoginScreen(): JSX.Element {
  return (
    <AppScreen style={commonStyle.container}>
      <LoginForm />
    </AppScreen>
  );
}
