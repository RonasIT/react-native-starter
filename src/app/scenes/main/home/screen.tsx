import React, { ReactElement } from 'react';
import { UsersList } from '@libs/users/features/list';
import { AppScreen } from '@shared/ui/ui-kit/screen';

export function HomeScreen(): ReactElement {
  return (
    <AppScreen testID='home-screen'>
      <UsersList />
    </AppScreen>
  );
}
