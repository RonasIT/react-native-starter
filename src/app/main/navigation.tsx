import { createBottomTabNavigator } from '@react-navigation/bottom-tabs';
import { useTranslation } from '@shared/i18n';
import { Icon } from '@shared/icon';
import { AppTabBar } from '@shared/tab-bar';
import { variables } from '@styles';
import React, { ReactElement } from 'react';
import { HomeScreen } from './home/screen';
import { ProfileScreen } from './profile/screen';

export type MainNavigationParams = {
  Home: undefined;
  Profile: undefined;
};

const Tab = createBottomTabNavigator<MainNavigationParams>();

export function MainNavigation(): ReactElement {
  const initialRouteName = 'Home';
  const translate = useTranslation('MAIN.NAVIGATION');

  return (
    <Tab.Navigator
      tabBar={(props) => <AppTabBar {...props} />}
      initialRouteName={initialRouteName}
      backBehavior='none'>
      <Tab.Screen
        name='Home'
        options={{
          tabBarIcon: ({ focused }) => (
            <Icon name='home' stroke={focused ? variables.color.primary : variables.color.white} />
          ),
          headerTitleAlign: 'left',
          title: translate('TEXT_HOME')
        }}
        component={HomeScreen}
      />
      <Tab.Screen
        name='Profile'
        options={{
          tabBarIcon: ({ focused }) => (
            <Icon name='user' stroke={focused ? variables.color.primary : variables.color.white} />
          ),
          headerTitleAlign: 'left',
          title: translate('TEXT_PROFILE')
        }}
        component={ProfileScreen}
      />
    </Tab.Navigator>
  );
}
