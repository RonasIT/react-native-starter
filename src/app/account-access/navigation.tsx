import { createStackNavigator } from '@react-navigation/stack';
import React, { ReactElement } from 'react';
import { appFacade } from '@app/facade';
import { useTranslation } from '@shared/i18n';
import { LoginScreen } from './login/screen';
import { PinScreen } from './pin';

export type AccountAccessNavigationParams = {
  Login: undefined;
  Pin: { isPinSet?: boolean };
};

const Stack = createStackNavigator<AccountAccessNavigationParams>();

export function AccountAccessNavigation(): ReactElement {
  const translate = useTranslation('ACCOUNT_ACCESS.NAVIGATION');
  const { isAuthenticated } = appFacade;

  return (
    <Stack.Navigator initialRouteName={isAuthenticated ? 'Pin' : 'Login'}>
      <Stack.Screen
        initialParams={{ isPinSet: isAuthenticated }}
        options={{ headerShown: false }}
        name='Pin'
        component={PinScreen}
      />
      <Stack.Screen
        name='Login'
        component={LoginScreen}
        options={{
          headerTitleAlign: 'left',
          title: translate('TEXT_LOG_IN')
        }}
      />
    </Stack.Navigator>
  );
}
