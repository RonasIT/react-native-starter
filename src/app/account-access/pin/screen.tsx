import { RouteProp } from '@react-navigation/native';
import React, { ReactElement, useEffect, useState } from 'react';
import { LocalAuthActions } from '@shared/local-auth';
import { localAuthService } from '@shared/local-auth/service';
import { PinForm } from '@shared/pin-form/component';
import { storeRef } from '@store';
import { AccountAccessNavigationParams } from '../navigation';

const pinLength = 4;

export function PinScreen(props: { route: RouteProp<AccountAccessNavigationParams, 'Pin'> }): ReactElement {
  const isPinSet = props?.route?.params?.isPinSet;
  const [inputValue, setInputValue] = useState('');
  const [confirmPin, setConfirmPin] = useState('');
  const [hasError, setHasError] = useState(false);
  const confirmMode = !isPinSet && inputValue.length === pinLength;

  const checkPinCode = async (): Promise<void> => {
    const isCorrect = await localAuthService.checkPin(inputValue);
    isCorrect ? storeRef.dispatch(LocalAuthActions.localAuthSuccess()) : setHasError(true);
  };

  const handleError = (): void => {
    setInputValue('');
    setConfirmPin('');
    setHasError(false);
  };

  useEffect(() => {
    if (inputValue.length === pinLength && isPinSet) {
      checkPinCode();
    }
    if (confirmMode && confirmPin.length === pinLength) {
      confirmPin === inputValue ? storeRef.dispatch(LocalAuthActions.setPin({ pin: confirmPin })) : setHasError(true);
    }
  }, [inputValue, confirmPin]);

  return (
    <PinForm
      hasError={hasError}
      handleError={handleError}
      pinLength={pinLength}
      value={confirmMode ? confirmPin : inputValue}
      onValueChange={confirmMode ? setConfirmPin : setInputValue}
      title={isPinSet ? 'Enter Pin' : confirmMode ? 'Confirm Pin' : 'Set Pin'}
      isPinSet={isPinSet}
    />
  );
}
