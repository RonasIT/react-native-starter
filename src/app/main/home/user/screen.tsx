import { RouteProp } from '@react-navigation/native';
import { useFormik } from 'formik';
import React, { useEffect } from 'react';
import { View } from 'react-native';
import { KeyboardAwareScrollView } from 'react-native-keyboard-aware-scroll-view';
import { HomeNavigationParams } from '@app/main/home/navigation';
import { AppButton } from '@shared/button';
import { useTranslation } from '@shared/i18n';
import { InputFormGroup } from '@shared/input-form-group';
import { AppScreen } from '@shared/screen';
import { AppText, TextTheme } from '@shared/text';
import { userAPI } from '@shared/user/api';
import { commonStyle, createStyles } from '@styles';
import { UserForm } from './shared/forms';

export function UserScreen(props: { route: RouteProp<HomeNavigationParams, 'User'> }): JSX.Element {
  const id = props.route.params?.id;
  const translate = useTranslation('MAIN.USER');
  const { useLazyGetQuery } = userAPI;
  const [loadUser, { data: user }] = useLazyGetQuery();

  useEffect(() => {
    if (id) {
      loadUser({ id });
    }
  }, []);

  useEffect(() => {
    if (user) {
      formik.setValues({
        email: user.email,
        name: user.name
      });
    }
  }, [user]);

  function formSubmitted(values: UserForm): void {
    console.log(values);
  }

  function deleteButtonPressed(): void {
    console.log('Delete');
  }

  const formik = useFormik({
    initialValues: new UserForm(),
    validationSchema: UserForm.validationSchema,
    onSubmit: formSubmitted
  });
  const { handleSubmit } = formik;

  return (
    <KeyboardAwareScrollView>
      <AppScreen style={commonStyle.container}>
        <AppText theme={TextTheme.LARGE} style={style.title}>
          {id ? translate('TEXT_EDIT') : translate('TEXT_CREATE')}
        </AppText>
        <InputFormGroup<UserForm>
          label={translate('TEXT_EMAIL')}
          name='email'
          autoCapitalize='none'
          keyboardType='email-address'
          formik={formik}
        />
        <InputFormGroup
          label={translate('TEXT_NAME')}
          name='name'
          formik={formik} />
        <View style={style.footer}>
          {id && (
            <AppButton
              title={translate('BUTTON_DELETE')}
              onPress={deleteButtonPressed}
              style={[style.button, style.deleteButton]}
              theme='secondary'
            />
          )}
          <AppButton
            isDisabled={!formik.isValid && formik.submitCount > 0}
            title={translate('BUTTON_SAVE')}
            onPress={() => handleSubmit()}
            style={style.button}
          />
        </View>
      </AppScreen>
    </KeyboardAwareScrollView>
  );
}

const style = createStyles({
  title: {
    marginVertical: '2rem'
  },
  footer: {
    marginTop: '1.5rem',
    flexDirection: 'row',
    justifyContent: 'space-between'
  },
  button: {
    flex: 1
  },
  deleteButton: {
    marginRight: '1rem'
  }
});
