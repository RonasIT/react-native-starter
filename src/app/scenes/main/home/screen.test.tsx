import { NavigationContainer } from '@react-navigation/native';
import { createStackNavigator } from '@react-navigation/stack';
import { fireEvent, render, RenderAPI, waitFor } from '@testing-library/react-native';
import { AxiosResponse } from 'axios';
import React from 'react';
import { act, ReactTestInstance } from 'react-test-renderer';
import { Observable, of } from 'rxjs';
import { apiService } from '@libs/shared/data-access/api-client';
import { navigationRef } from '@libs/shared/features/navigation';
import { userPaginationResponse } from '@tests/fixtures';
import { scrollDownEventData, TestRootComponent } from '@tests/helpers';
import { HomeScreen } from './screen';
import SpyInstance = jest.SpyInstance;

describe('Home screen', () => {
  let component: RenderAPI;
  let usersList: ReactTestInstance;
  let createUserButton: ReactTestInstance;
  let navigateSpy: SpyInstance;
  const searchUsersSpy = jest.spyOn(apiService.httpClient, 'request');

  function initComponent(): RenderAPI {
    const { Screen, Navigator } = createStackNavigator();

    return render(
      <TestRootComponent>
        <NavigationContainer ref={navigationRef}>
          <Navigator>
            <Screen name='Home' component={HomeScreen} />
          </Navigator>
        </NavigationContainer>
      </TestRootComponent>
    );
  }

  beforeAll(() => {
    jest.spyOn(apiService.httpClient, 'request').mockImplementation((config) => {
      if (config.method === 'get' && config.url === '/users') {
        return of({ data: userPaginationResponse }) as Observable<AxiosResponse>;
      }
    });
  });

  beforeEach(() => {
    component = initComponent();
    usersList = component.getByTestId('users-list');
    createUserButton = component.getByTestId('create-user-button');
    navigateSpy = jest.spyOn(navigationRef.current, 'navigate');
  });

  it('should match the snapshot', () => {
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render list items', () => {
    const listItems = component.getAllByTestId('user-item');
    expect(listItems).toHaveLength(userPaginationResponse.data.length);
  });

  it('should render the button "Create user"', () => {
    expect(createUserButton).toBeDefined();
  });

  it('should load more items after the list end reached', async () => {
    fireEvent.scroll(usersList, scrollDownEventData);

    await waitFor(() => {
      expect(searchUsersSpy).toHaveBeenCalledWith(
        expect.objectContaining({
          method: 'get',
          url: '/users',
          params: { page: 2 }
        })
      );
    });
  });

  it('should load the first page of users on the list refresh', async () => {
    const { refreshControl } = usersList.props;
    act(() => {
      refreshControl.props.onRefresh();
    });

    await waitFor(() => {
      expect(searchUsersSpy).toHaveBeenCalledWith(
        expect.objectContaining({
          method: 'get',
          url: '/users',
          params: { page: 1 }
        })
      );
    });
  });

  it('should navigate to empty user page by pressing the button "Create user"', async () => {
    fireEvent.press(createUserButton);

    expect(navigateSpy).toHaveBeenCalledWith('User');
  });

  it('should navigate to user page by pressing an item', async () => {
    const listItems = component.getAllByTestId('user-item');
    const itemIndex = Math.floor(Math.random() * listItems.length);
    fireEvent.press(listItems[itemIndex]);

    expect(navigateSpy).toHaveBeenCalledWith('User', { id: userPaginationResponse.data[itemIndex].id });
  });
});
