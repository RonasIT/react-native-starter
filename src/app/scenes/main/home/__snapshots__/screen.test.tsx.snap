// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home screen should match the snapshot 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    style={
      {
        "flex": 1,
      }
    }
  >
    <View
      style={
        [
          {
            "flex": 1,
          },
          undefined,
        ]
      }
    >
      <View
        style={
          [
            {
              "backgroundColor": "rgb(242, 242, 242)",
              "flex": 1,
            },
            undefined,
          ]
        }
      >
        <View
          collapsable={false}
          pointerEvents="box-none"
          style={
            {
              "zIndex": 1,
            }
          }
        >
          <View
            accessibilityElementsHidden={false}
            importantForAccessibility="auto"
            onLayout={[Function]}
            pointerEvents="box-none"
            style={null}
          >
            <View
              collapsable={false}
              pointerEvents="box-none"
              style={
                {
                  "bottom": 0,
                  "left": 0,
                  "opacity": 1,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "zIndex": 0,
                }
              }
            >
              <View
                collapsable={false}
                style={
                  {
                    "backgroundColor": "rgb(255, 255, 255)",
                    "borderBottomColor": "rgb(216, 216, 216)",
                    "flex": 1,
                    "shadowColor": "rgb(216, 216, 216)",
                    "shadowOffset": {
                      "height": 0.5,
                      "width": 0,
                    },
                    "shadowOpacity": 0.85,
                    "shadowRadius": 0,
                  }
                }
              />
            </View>
            <View
              collapsable={false}
              pointerEvents="box-none"
              style={
                {
                  "height": 44,
                  "maxHeight": undefined,
                  "minHeight": undefined,
                  "opacity": undefined,
                  "transform": undefined,
                }
              }
            >
              <View
                pointerEvents="none"
                style={
                  {
                    "height": 0,
                  }
                }
              />
              <View
                pointerEvents="box-none"
                style={
                  {
                    "alignItems": "stretch",
                    "flex": 1,
                    "flexDirection": "row",
                  }
                }
              >
                <View
                  collapsable={false}
                  pointerEvents="box-none"
                  style={
                    {
                      "alignItems": "flex-start",
                      "flexBasis": 0,
                      "flexGrow": 1,
                      "justifyContent": "center",
                      "marginStart": 0,
                      "opacity": 1,
                    }
                  }
                />
                <View
                  collapsable={false}
                  pointerEvents="box-none"
                  style={
                    {
                      "justifyContent": "center",
                      "marginHorizontal": 16,
                      "maxWidth": 718,
                      "opacity": 1,
                    }
                  }
                >
                  <Text
                    accessibilityRole="header"
                    aria-level="1"
                    collapsable={false}
                    numberOfLines={1}
                    onLayout={[Function]}
                    style={
                      {
                        "color": "rgb(28, 28, 30)",
                        "fontSize": 17,
                        "fontWeight": "600",
                      }
                    }
                  >
                    Home
                  </Text>
                </View>
                <View
                  collapsable={false}
                  pointerEvents="box-none"
                  style={
                    {
                      "alignItems": "flex-end",
                      "flexBasis": 0,
                      "flexGrow": 1,
                      "justifyContent": "center",
                      "marginEnd": 0,
                      "opacity": 1,
                    }
                  }
                />
              </View>
            </View>
          </View>
        </View>
        <RNSScreenContainer
          onLayout={[Function]}
          style={
            {
              "flex": 1,
            }
          }
        >
          <RNSScreen
            activityState={2}
            collapsable={false}
            forwardedRef={[Function]}
            gestureResponseDistance={
              {
                "bottom": -1,
                "end": -1,
                "start": -1,
                "top": -1,
              }
            }
            pointerEvents="box-none"
            style={
              {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          >
            <View
              collapsable={false}
              style={
                {
                  "opacity": 1,
                }
              }
            />
            <View
              accessibilityElementsHidden={false}
              closing={false}
              gestureVelocityImpact={0.3}
              importantForAccessibility="auto"
              onClose={[Function]}
              onGestureBegin={[Function]}
              onGestureCanceled={[Function]}
              onGestureEnd={[Function]}
              onOpen={[Function]}
              onTransition={[Function]}
              pointerEvents="box-none"
              style={
                [
                  {
                    "display": "flex",
                    "overflow": undefined,
                  },
                  {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  },
                ]
              }
              transitionSpec={
                {
                  "close": {
                    "animation": "spring",
                    "config": {
                      "damping": 500,
                      "mass": 3,
                      "overshootClamping": true,
                      "restDisplacementThreshold": 10,
                      "restSpeedThreshold": 10,
                      "stiffness": 1000,
                    },
                  },
                  "open": {
                    "animation": "spring",
                    "config": {
                      "damping": 500,
                      "mass": 3,
                      "overshootClamping": true,
                      "restDisplacementThreshold": 10,
                      "restSpeedThreshold": 10,
                      "stiffness": 1000,
                    },
                  },
                }
              }
            >
              <View
                collapsable={false}
                pointerEvents="box-none"
                style={
                  {
                    "flex": 1,
                  }
                }
              >
                <View
                  collapsable={false}
                  forwardedRef={[Function]}
                  handlerTag={1}
                  handlerType="PanGestureHandler"
                  needsOffscreenAlphaCompositing={false}
                  onGestureHandlerEvent={[Function]}
                  onGestureHandlerStateChange={[Function]}
                  style={
                    {
                      "flex": 1,
                      "transform": [
                        {
                          "translateX": 0,
                        },
                        {
                          "translateX": 0,
                        },
                      ],
                    }
                  }
                >
                  <View
                    pointerEvents="box-none"
                    style={
                      [
                        {
                          "flex": 1,
                          "overflow": "hidden",
                        },
                        [
                          {
                            "backgroundColor": "rgb(242, 242, 242)",
                          },
                          undefined,
                        ],
                      ]
                    }
                  >
                    <View
                      style={
                        {
                          "flex": 1,
                          "flexDirection": "column-reverse",
                        }
                      }
                    >
                      <View
                        style={
                          {
                            "flex": 1,
                          }
                        }
                      >
                        <View
                          style={
                            [
                              {
                                "flex": 1,
                              },
                              {},
                            ]
                          }
                          testID="home-screen"
                        >
                          <RCTScrollView
                            ListEmptyComponent={<ItemsListEmptyState />}
                            ListFooterComponent={false}
                            ListHeaderComponent={
                              <AppButton
                                label="[missing "en.USERS.LIST.BUTTON_CREATE_USER" translation]"
                                onPress={[Function]}
                                style={
                                  {
                                    "marginBottom": 16,
                                  }
                                }
                                testID="create-user-button"
                              />
                            }
                            contentContainerStyle={
                              [
                                {
                                  "minHeight": "100%",
                                  "paddingBottom": 50,
                                },
                                {
                                  "paddingHorizontal": 24,
                                },
                              ]
                            }
                            data={
                              [
                                User {
                                  "createdAt": undefined,
                                  "deletedAt": undefined,
                                  "email": "test@test.com",
                                  "gender": "female",
                                  "id": 2358991,
                                  "name": "Devak Kakkar",
                                  "status": "inactive",
                                  "updatedAt": undefined,
                                  Symbol(immer-draftable): true,
                                },
                                User {
                                  "createdAt": undefined,
                                  "deletedAt": undefined,
                                  "email": "test2@test.com",
                                  "gender": "female",
                                  "id": 2358990,
                                  "name": "Chiranjeev Nehru DVM",
                                  "status": "active",
                                  "updatedAt": undefined,
                                  Symbol(immer-draftable): true,
                                },
                                User {
                                  "createdAt": undefined,
                                  "deletedAt": undefined,
                                  "email": "test3@test.com",
                                  "gender": "male",
                                  "id": 2358989,
                                  "name": "Dayaananda Bhat",
                                  "status": "active",
                                  "updatedAt": undefined,
                                  Symbol(immer-draftable): true,
                                },
                              ]
                            }
                            getItem={[Function]}
                            getItemCount={[Function]}
                            keyExtractor={[Function]}
                            onContentSizeChange={[Function]}
                            onEndReached={[Function]}
                            onLayout={[Function]}
                            onMomentumScrollBegin={[Function]}
                            onMomentumScrollEnd={[Function]}
                            onScroll={[Function]}
                            onScrollBeginDrag={[Function]}
                            onScrollEndDrag={[Function]}
                            refreshControl={
                              <AppRefreshControl
                                onRefresh={[Function]}
                                refreshing={false}
                              />
                            }
                            removeClippedSubviews={false}
                            renderItem={[Function]}
                            scrollEventThrottle={50}
                            stickyHeaderIndices={[]}
                            testID="users-list"
                            viewabilityConfigCallbackPairs={[]}
                          >
                            <RCTRefreshControl />
                            <View>
                              <View
                                onLayout={[Function]}
                              >
                                <View
                                  accessibilityRole="button"
                                  accessibilityState={
                                    {
                                      "busy": undefined,
                                      "checked": undefined,
                                      "disabled": undefined,
                                      "expanded": undefined,
                                      "selected": undefined,
                                    }
                                  }
                                  accessibilityValue={
                                    {
                                      "max": undefined,
                                      "min": undefined,
                                      "now": undefined,
                                      "text": undefined,
                                    }
                                  }
                                  accessible={true}
                                  collapsable={false}
                                  focusable={true}
                                  onClick={[Function]}
                                  onLayout={[Function]}
                                  onResponderGrant={[Function]}
                                  onResponderMove={[Function]}
                                  onResponderRelease={[Function]}
                                  onResponderTerminate={[Function]}
                                  onResponderTerminationRequest={[Function]}
                                  onStartShouldSetResponder={[Function]}
                                  style={
                                    {
                                      "alignItems": "center",
                                      "backgroundColor": "#26a0f8",
                                      "borderRadius": 12,
                                      "flexDirection": "row",
                                      "justifyContent": "center",
                                      "marginBottom": 16,
                                      "minWidth": 90,
                                      "opacity": 1,
                                      "paddingHorizontal": 15,
                                      "paddingVertical": 13,
                                      "width": "100%",
                                    }
                                  }
                                  testID="create-user-button"
                                >
                                  <Text
                                    numberOfLines={1}
                                    style={
                                      [
                                        {
                                          "backgroundColor": "transparent",
                                          "color": "#20303C",
                                          "textAlign": "left",
                                        },
                                        undefined,
                                        undefined,
                                        undefined,
                                        undefined,
                                        undefined,
                                        undefined,
                                        undefined,
                                        undefined,
                                        [
                                          {
                                            "backgroundColor": "transparent",
                                            "flexDirection": "row",
                                            "fontFamily": "System",
                                            "fontSize": 16,
                                            "fontWeight": "400",
                                            "lineHeight": 24,
                                          },
                                          {
                                            "color": "#FFFFFF",
                                          },
                                          undefined,
                                          undefined,
                                          [
                                            {
                                              "fontFamily": "SFProTextSemiBold",
                                              "fontSize": 18.4,
                                              "fontWeight": "600",
                                              "lineHeight": 30,
                                              "textAlign": "center",
                                            },
                                            {
                                              "color": "#FFFFFF",
                                            },
                                            undefined,
                                          ],
                                        ],
                                      ]
                                    }
                                    testID="create-user-button.label"
                                  >
                                    [missing "en.USERS.LIST.BUTTON_CREATE_USER" translation]
                                  </Text>
                                </View>
                              </View>
                              <View
                                onFocusCapture={[Function]}
                                onLayout={[Function]}
                                style={null}
                              >
                                <View
                                  accessibilityState={
                                    {
                                      "busy": undefined,
                                      "checked": undefined,
                                      "disabled": undefined,
                                      "expanded": undefined,
                                      "selected": undefined,
                                    }
                                  }
                                  accessibilityValue={
                                    {
                                      "max": undefined,
                                      "min": undefined,
                                      "now": undefined,
                                      "text": undefined,
                                    }
                                  }
                                  accessible={true}
                                  collapsable={false}
                                  focusable={true}
                                  onClick={[Function]}
                                  onResponderGrant={[Function]}
                                  onResponderMove={[Function]}
                                  onResponderRelease={[Function]}
                                  onResponderTerminate={[Function]}
                                  onResponderTerminationRequest={[Function]}
                                  onStartShouldSetResponder={[Function]}
                                  style={
                                    {
                                      "borderBottomColor": "#FFFFFF",
                                      "borderBottomWidth": 2,
                                      "opacity": 1,
                                      "paddingVertical": 8,
                                    }
                                  }
                                  testID="user-item"
                                >
                                  <Text
                                    numberOfLines={1}
                                    style={
                                      [
                                        {
                                          "backgroundColor": "transparent",
                                          "color": "#20303C",
                                          "textAlign": "left",
                                        },
                                        undefined,
                                        undefined,
                                        undefined,
                                        undefined,
                                        undefined,
                                        undefined,
                                        undefined,
                                        undefined,
                                        [
                                          {
                                            "fontFamily": "SFProDisplayRegular",
                                            "fontSize": 22.4,
                                            "lineHeight": 24.64,
                                          },
                                          {
                                            "color": "#FFFFFF",
                                          },
                                          undefined,
                                        ],
                                      ]
                                    }
                                  >
                                    #
                                    2358991
                                    : 
                                    Devak Kakkar
                                  </Text>
                                  <Text
                                    numberOfLines={1}
                                    style={
                                      [
                                        {
                                          "backgroundColor": "transparent",
                                          "color": "#20303C",
                                          "textAlign": "left",
                                        },
                                        undefined,
                                        undefined,
                                        undefined,
                                        undefined,
                                        undefined,
                                        undefined,
                                        undefined,
                                        undefined,
                                        [
                                          {
                                            "fontFamily": "SFProTextRegular",
                                            "fontSize": 16,
                                            "lineHeight": 29.44,
                                          },
                                          {
                                            "color": "#FFFFFF",
                                          },
                                          undefined,
                                        ],
                                      ]
                                    }
                                  >
                                    Email: 
                                    test@test.com
                                  </Text>
                                </View>
                              </View>
                              <View
                                onFocusCapture={[Function]}
                                onLayout={[Function]}
                                style={null}
                              >
                                <View
                                  accessibilityState={
                                    {
                                      "busy": undefined,
                                      "checked": undefined,
                                      "disabled": undefined,
                                      "expanded": undefined,
                                      "selected": undefined,
                                    }
                                  }
                                  accessibilityValue={
                                    {
                                      "max": undefined,
                                      "min": undefined,
                                      "now": undefined,
                                      "text": undefined,
                                    }
                                  }
                                  accessible={true}
                                  collapsable={false}
                                  focusable={true}
                                  onClick={[Function]}
                                  onResponderGrant={[Function]}
                                  onResponderMove={[Function]}
                                  onResponderRelease={[Function]}
                                  onResponderTerminate={[Function]}
                                  onResponderTerminationRequest={[Function]}
                                  onStartShouldSetResponder={[Function]}
                                  style={
                                    {
                                      "borderBottomColor": "#FFFFFF",
                                      "borderBottomWidth": 2,
                                      "opacity": 1,
                                      "paddingVertical": 8,
                                    }
                                  }
                                  testID="user-item"
                                >
                                  <Text
                                    numberOfLines={1}
                                    style={
                                      [
                                        {
                                          "backgroundColor": "transparent",
                                          "color": "#20303C",
                                          "textAlign": "left",
                                        },
                                        undefined,
                                        undefined,
                                        undefined,
                                        undefined,
                                        undefined,
                                        undefined,
                                        undefined,
                                        undefined,
                                        [
                                          {
                                            "fontFamily": "SFProDisplayRegular",
                                            "fontSize": 22.4,
                                            "lineHeight": 24.64,
                                          },
                                          {
                                            "color": "#FFFFFF",
                                          },
                                          undefined,
                                        ],
                                      ]
                                    }
                                  >
                                    #
                                    2358990
                                    : 
                                    Chiranjeev Nehru DVM
                                  </Text>
                                  <Text
                                    numberOfLines={1}
                                    style={
                                      [
                                        {
                                          "backgroundColor": "transparent",
                                          "color": "#20303C",
                                          "textAlign": "left",
                                        },
                                        undefined,
                                        undefined,
                                        undefined,
                                        undefined,
                                        undefined,
                                        undefined,
                                        undefined,
                                        undefined,
                                        [
                                          {
                                            "fontFamily": "SFProTextRegular",
                                            "fontSize": 16,
                                            "lineHeight": 29.44,
                                          },
                                          {
                                            "color": "#FFFFFF",
                                          },
                                          undefined,
                                        ],
                                      ]
                                    }
                                  >
                                    Email: 
                                    test2@test.com
                                  </Text>
                                </View>
                              </View>
                              <View
                                onFocusCapture={[Function]}
                                onLayout={[Function]}
                                style={null}
                              >
                                <View
                                  accessibilityState={
                                    {
                                      "busy": undefined,
                                      "checked": undefined,
                                      "disabled": undefined,
                                      "expanded": undefined,
                                      "selected": undefined,
                                    }
                                  }
                                  accessibilityValue={
                                    {
                                      "max": undefined,
                                      "min": undefined,
                                      "now": undefined,
                                      "text": undefined,
                                    }
                                  }
                                  accessible={true}
                                  collapsable={false}
                                  focusable={true}
                                  onClick={[Function]}
                                  onResponderGrant={[Function]}
                                  onResponderMove={[Function]}
                                  onResponderRelease={[Function]}
                                  onResponderTerminate={[Function]}
                                  onResponderTerminationRequest={[Function]}
                                  onStartShouldSetResponder={[Function]}
                                  style={
                                    {
                                      "borderBottomColor": "#FFFFFF",
                                      "borderBottomWidth": 2,
                                      "opacity": 1,
                                      "paddingVertical": 8,
                                    }
                                  }
                                  testID="user-item"
                                >
                                  <Text
                                    numberOfLines={1}
                                    style={
                                      [
                                        {
                                          "backgroundColor": "transparent",
                                          "color": "#20303C",
                                          "textAlign": "left",
                                        },
                                        undefined,
                                        undefined,
                                        undefined,
                                        undefined,
                                        undefined,
                                        undefined,
                                        undefined,
                                        undefined,
                                        [
                                          {
                                            "fontFamily": "SFProDisplayRegular",
                                            "fontSize": 22.4,
                                            "lineHeight": 24.64,
                                          },
                                          {
                                            "color": "#FFFFFF",
                                          },
                                          undefined,
                                        ],
                                      ]
                                    }
                                  >
                                    #
                                    2358989
                                    : 
                                    Dayaananda Bhat
                                  </Text>
                                  <Text
                                    numberOfLines={1}
                                    style={
                                      [
                                        {
                                          "backgroundColor": "transparent",
                                          "color": "#20303C",
                                          "textAlign": "left",
                                        },
                                        undefined,
                                        undefined,
                                        undefined,
                                        undefined,
                                        undefined,
                                        undefined,
                                        undefined,
                                        undefined,
                                        [
                                          {
                                            "fontFamily": "SFProTextRegular",
                                            "fontSize": 16,
                                            "lineHeight": 29.44,
                                          },
                                          {
                                            "color": "#FFFFFF",
                                          },
                                          undefined,
                                        ],
                                      ]
                                    }
                                  >
                                    Email: 
                                    test3@test.com
                                  </Text>
                                </View>
                              </View>
                            </View>
                          </RCTScrollView>
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </RNSScreen>
        </RNSScreenContainer>
      </View>
    </View>
  </View>
</RNCSafeAreaProvider>
`;
